lib/oli/analytics/datashop_export_worker.ex

```diff
@@ -121,4 +89,45 @@ defmodule Oli.Analytics.DatashopExportWorker do
 
     result
   end
+
+  defp write_to_file(temp_file_name, project, section_ids) do
+    first_chunk = Datashop.content_prefix()
+    last_chunk = Datashop.content_suffix()
+
+    batch_size = Datashop.max_record_size()
+    total = Datashop.count(section_ids)
+    batch_count = ceil(total / batch_size)
+
+    if batch_count != 0 do
+      Enum.reduce(1..batch_count, 0, fn batch_index, offset ->
+        # notify subscribers that a new batch has started
+        Broadcaster.broadcast_datashop_export_batch_started(
+          project.slug,
+          batch_index,
+          batch_count
+        )
+
+        Datashop.build_context(project.id, section_ids)
```

Comment:
The call to `Datashop.build_context` can be done one time, outside of this `Enum.reduce`.  That will limit the number of queries that the Datashop export makes

Reply:
Makes sense! The last commit includes these changes :)